sktree/datasets/hyppo.py

```diff
@@ -123,36 +147,280 @@ def make_trunk_classification(
     else:
         cov = np.identity(n_dim)
 
-    if mix < 0 or mix > 1:
-        raise ValueError("Mix must be between 0 and 1.")
-
     if n_dim > 1000:
         method = "cholesky"
     else:
         method = "svd"
 
-    if mix == 0:
+    if simulation == "trunk":
         X = np.vstack(
             (
                 rng.multivariate_normal(mu_0, cov, n_samples // 2, method=method),
                 rng.multivariate_normal(mu_1, cov, n_samples // 2, method=method),
             )
         )
-    else:
+    elif simulation == "trunk_overlap":
+        X = np.vstack(
+            (
+                rng.multivariate_normal(np.zeros(n_dim), cov, n_samples // 2, method=method),
+                rng.multivariate_normal(np.zeros(n_dim), cov, n_samples // 2, method=method),
+            )
+        )
+    elif simulation == "trunk_mix":
+        mixture_idx = rng.choice(2, n_samples // 2, replace=True, shuffle=True, p=None)
+        X_mixture = np.zeros((n_samples // 2, len(mu_1)))
+        for idx in range(n_samples // 2):
+            if mixture_idx[idx] == 1:
+                X_sample = rng.multivariate_normal(mu_1, cov * (2 / 3) ** 2, 1, method=method)
+            else:
+                X_sample = rng.multivariate_normal(mu_0, cov * (2 / 3) ** 2, 1, method=method)
+            X_mixture[idx, :] = X_sample
+
+        X = np.vstack(
+            (
+                rng.multivariate_normal(np.zeros(n_dim), cov, n_samples // 2, method=method),
+                X_mixture,
+            )
+        )
+    elif simulation in MARRON_WAND_SIMS.keys():
+        mixture_idx = rng.choice(
+            MARRON_WAND_SIMS[simulation][0],
+            n_samples // 2,
+            replace=True,
+            shuffle=True,
+            p=MARRON_WAND_SIMS[simulation][1],
+        )
+        G = np.zeros((n_samples // 2, len(w)))
+        for idx in range(n_samples // 2):
+            G[idx, :] = MarronWandSims(
+                n_dim=n_dim,
+                cov=cov,
+                method=method,
+                mix_id=mixture_idx[idx],
+                rng=rng,
+                max_mix_id=MARRON_WAND_SIMS[simulation][0],
+            )(simulation)
+
         X = np.vstack(
             (
                 rng.multivariate_normal(np.zeros(n_dim), cov, n_samples // 2, method=method),
-                (1 - mix) * rng.multivariate_normal(mu_1, cov, n_samples // 2, method=method)
-                + mix * rng.multivariate_normal(mu_0, cov, n_samples // 2, method=method),
+                (1 - w)
+                * rng.multivariate_normal(np.zeros(n_dim), cov, n_samples // 2, method=method)
+                + w * G,
             )
         )
+    else:
+        raise ValueError(
+            f"Simulation must be trunk, trunk_overlap, trunk_mix, {MARRON_WAND_SIMS.keys()}"
+        )
+
     y = np.concatenate((np.zeros(n_samples // 2), np.ones(n_samples // 2)))
 
     if return_params:
         return X, y, [mu_0, mu_1], [cov, cov]
```

Comment:
Technically, the mu and covariance are not easily returned anymore. I'm wondering if we should wrap them in an object then instead?

Reply:
I'm not sure, what do you think we need to return in `return_params`?

Reply:
I guess we use `return_params` to get information about the data generating model as well as possibly parameters of the DGM.

Is it easy to wrap these as a subclass for `rv_continuous` class (https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.html#scipy.stats.rv_continuous)? Then we can just return the instantiated object. Though I'm not familiar with that API, and how we can recover the exact parametrization of a specific MW simulation.

Reply:
@sampan501 if there's no way to address this, then should we just error out on `return_params=True` if simulation is not trunk?

I think we just want the DGM to be returned, so perhaps it would make sense to just return a callable such that
`returned_model.sample(n_samples)` can generate new samples of data. For `rng.multivariate_normal` or if we can get an API in the framework of `rv_continuous`, then we can do things like:

```
test = multivariate_normal(mean=2.5, cov=0.5)
print(test)
print(test.mean)
print(test.cov)
print(test.rvs(size=(10,)))
```

Reply:
we could just return `X`, `Y`, `norm_params`, and `G` for the Marron and Wand Sims right?

Reply:
The only think additional is G, which are the observations from the mixed Gaussian

Reply:
Yeah that sounds reasonable for now. Can you update the docstring accordingly?

Reply:
https://github.com/neurodata/scikit-tree/pull/203/commits/6688725b70666c5b86b3d4c87f1cba316340eeca